### PR TITLE
First step of fixing the problem with generating Gradle/Bintray URLs

### PR DESCRIPTION
**Pull request by wwilk**
First step of #478. It solves the problem for Powermock, so that they could use "powermock-{VERSION}" in the release notes. When it's merged I will raise a PR in powermock to fix this issue.

### CODE REVIEW

**Comment by mockitoguy** [comment]
The powermock use case is already handled. We can do something like that:

```
shipkit {
  releaseNotes.publicationRepository = "https://bintray.com/powermock/maven/powermock/powermock-$version"
}
```

Why do we need the "{VERSION}" placeholder?

**Comment by wwilk** [comment]
Placeholder instead of concatenation should be easier to use and more self-explanatory. If you see "http://https://bintray.com/powermock/maven/powermock/powermock-" in the code, you don't really know that version will be concatenated to it. It's quite clear when you see {VERSION} placeholder. I thought that maybe in the future we would be able to add more placeholders. 

I'm not so sure about it, but I don't like current concatenation
```
"publicationRepository` + "/" + version"
```
because if you want to use it, you have to check if "/" is added in the Shipkit code or not. In case of placeholder you don't really care about this.

What do you think @mockitoguy ?

**Comment by mockitoguy** [comment]
My gut feel is that it is an unnecessary complexity.

1. Currently the user can do this:
```
shipkit {
  releaseNotes.publicationRepository = "https://bintray.com/powermock/maven/powermock/powermock-$version"
}
```

2. This PR makes it so that the user can do that:
```
shipkit {
  releaseNotes.publicationRepository = "https://bintray.com/powermock/maven/powermock/powermock-{VERSION}"
}
```

The ROI is minimal, if any. Both look clean to me.

Approach 1) is based on well known GString variable replacement + well known project property "version". Approach 2) introduces a brand new style of token replacement that needs to be implemented, documented and maintained. Users might even try to use "{VERSION}" keyword in other Shipkit properties...

That's my honest feedback :) I'm curious what do other contributors think?

**Comment by epeee** [comment]
I'd vote for Approach 1 (since it is well known and we do not have extra effort).

**Comment by wwilk** [comment]
OK let's keep concatenation, but I think it would be better to avoid adding a version with slash "/". Otherwise it's not really possible to have a version like "powermock-1.2.3", it would be "powermock-/1.2.3". So I would like to still change this line:
https://github.com/mockito/shipkit/blob/514ac7a43af0e3513a83f7518221086b203d4b45/subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/format/DetailedFormatter.java#L82
to "publicationRepository + version". And also encapsulating this part in a separate class. 
OK with you guys? @mockitoguy @epeee ?
Btw because of that line, user can't really do sth like "powermock-$version". It would result in "powermock-1.2.3/1.2.3"

**Comment by mockitoguy** [comment]
Oh, I didn't see that it wouldn't work!

Let's simplify this. Let's remove version concat completely and have the formatter just use the "publicationRepository".

This way, the following will work:

```
//powermock
shipkit {
  releaseNotes.publicationRepository = "https://bintray.com/powermock/maven/powermock/powermock-$version"
}

//shipkit
shipkit {
  releaseNotes.publicationRepository = "https://plugins.gradle.org/plugin/org.shipkit.java/$version"
}
```

File: subprojects/shipkit/src/main/groovy/org/shipkit/gradle/configuration/ShipkitConfiguration.java [hunk 1]
```diff
@@ -255,10 +255,13 @@ public void setIgnoreCommitsContaining(Collection<String> commitMessageParts) {
 
         /**
          * Set the Publication Repository where we look for your published binary.
```
Comment (mockitoguy) [inline subprojects/shipkit/src/main/groovy/org/shipkit/gradle/configuration/ShipkitConfiguration.java]: Let's write what do we use it for:
 - release notes generation
 - getting binaries for comparing with previous version

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/comparison/artifact/DefaultArtifactUrlResolverFactory.java [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 public class DefaultArtifactUrlResolverFactory {
 
+    private static final String VERSION_PLACEHOLDER = "{VERSION}";
```
Comment (mockitoguy) [inline subprojects/shipkit/src/main/groovy/org/shipkit/internal/comparison/artifact/DefaultArtifactUrlResolverFactory.java]: Unused constant

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/tasks/DownloadPreviousPublications.java [hunk 3]
```diff
@@ -25,7 +25,9 @@ private void downloadRemoteFile(String remoteUrl, File localFile) {
             IOUtil.downloadToFile(remoteUrl, localFile);
         } catch (Exception e) {
             LOG.lifecycle("  Unable to download, ignoring. Run with '-d' for stack trace. Url: {}", remoteUrl);
-            LOG.debug("Unable to download, ignoring", e);
+            LOG.debug("Unable to download, ignoring." +
```
Comment (mockitoguy) [inline subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/tasks/DownloadPreviousPublications.java]: This should probably be a "lifecycle" message, right?

Suggestion: 
 - lifecycle message with general info + a hint how to resolve the issue
 - debug message with stack trace

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/format/DetailedFormatter.java [hunk 4]
```diff
@@ -46,7 +47,7 @@ public String formatReleaseNotes(Collection<ReleaseNotesData> data) {
         for (ReleaseNotesData d : data) {
             sb.append(header(d.getVersion(), d.getDate(), emphasizeVersion));
             String vcsCommitsLink = MessageFormat.format(vcsCommitsLinkTemplate, d.getPreviousVersionVcsTag(), d.getVcsTag());
-            sb.append(releaseSummary(d.getVersion(), d.getDate(), d.getContributions(), contributors, vcsCommitsLink, publicationRepository));
```
Comment (mockitoguy) [inline subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/format/DetailedFormatter.java]: Nice! Thank you for removing dead parameter.

**Pull request merged**